include/ev3_sensors/ev3_gyro.h

```diff
@@ -3,30 +3,191 @@
+ *         On some sensors the value will overflow, on other ones
+ *         the value will stay stuck at +32767 or -32768 (see ev3dev wiki)
+ */
+typedef enum EV3GyroAngleMode {
+    /**
+     * @brief Read the angle measured by the sensor using the GYRO-ANG mode.
+     * The measurement will be done along the axis marked on the sensor.
+     * Clockwise rotation (when looking on the sensor top) will increase the reported values.
+     */
+    EV3GyroNormalAngle,
+
+    /**
+     * @brief Read the angle measured by the sensor using the GYRO-G&A mode.
+     * Use this mode if you want to read both angle and rate at once.
+     * The measurement will be done along the axis marked on the sensor.
+     * Clockwise rotation (when looking on the sensor top) will increase the reported values.
+     * @remark On N2/N3 sensors, switching out of this mode will take
+     *         a long time (~3 seconds), because the sensor has to time out
+     *         and reboot. On newer sensors, this shouldn't be an issue.
+     */
+    EV3GyroNormalAngleWithRate,
+
+    /**
+     * @brief Read the angle along an alternative axis using the TILT-ANG mode.
+     * The measurement will be done along the cable axis.
+     * Clockwise rotation (when looking against the cable plug) will increase the reported values.
+     * @remark This mode is only available in N4+ sensors.
+     */
+    EV3GyroTiltAngle,
+} EV3GyroAngleMode;
+
+/**
+ * @brief Rate reading modes.
+ */
+typedef enum EV3GyroRateMode {
+    /**
+     * @brief Read the rotation rate measured by the sensor using the GYRO-RATE mode.
+     * The measurement will be done along the axis marked on the sensor.
+     * Clockwise rotation (when looking on the sensor top) will report positive values.
+     * @remark This mode supports only +- 440 °/s rotational rates.
+     *         For faster rotations, try using the GYRO-FAS mode.
+     */
+    EV3GyroNormalRate,
+
+    /**
+     * @brief Read the fast rotation rate measured by the sensor using the GYRO-FAS mode.
+     * Use this mode if you want to read both angle and rate at once.
+     * The measurement will be done along the axis marked on the sensor.
+     * Clockwise rotation (when looking on the sensor top) will report positive values.
+     * @remark This mode supports +- 2000 °/s rotational rates. As a
+     *         tradeoff, it likely has lower precision compared to the GYRO-RATE mode.
+     */
+    EV3GyroFastRate,
+
+    /**
+     * @brief Read both the angle and rate measurements using the GYRO-G&A mode.
+     * The measurement will be done along the axis marked on the sensor.
+     * Clockwise rotation (when looking on the sensor top) will report positive values.
+     * @remark On N2/N3 sensors, switching out of this mode will take
+     *         a long time (~3 seconds), because the sensor has to time out
+     *         and reboot. On newer sensors, this shouldn't be an issue.
+     */
+    EV3GyroNormalRateWithAngle,
```

Comment:
I like the proposed API for the gyro sensor.
However, I'm having some troubles to convince myself about the existence of two enum items that represent the same mode: `EV3GyroNormalAngleWithRate` and `EV3GyroNormalRateWithAngle`.

What I fear is that someone may think that one equals the other, and write something like this:
```c
int angle = ReadEV3GyroSensorAngle(IN_1, EV3GyroNormalAngleWithRate);
int rate = ReadEV3GyroSensorRate(IN_1, EV3GyroNormalAngleWithRate); // wrong!
```

The second line would not read the rate, but the fast rate.

A possible (hacky) solution could be to declare the two `EV3GyroNormalAngleWithRate` and `EV3GyroNormalRateWithAngle` in the same position of the enums, so that they both have the same value. For example, we could declare them as first, so that they both have the value 0.

The example program would still be _wrong_, but at least the it would behave as expected.

Reply:
Yup, that feels a bit strange. Do you think replacing the names with `EV3GyroCombinedAngle` or `EV3GyroFusedAngle` would help? "Combining" or "fusing" could be understood differently, like combining two axes, which isn't the case here, but I don't know a better word for it.

Reply:
I have opted to using `EV3GyroInterleavedAngle` and `EV3GyroInterleavedRate` as that captures the intended use case - calling both functions in quick succession.

Reply:
I think it's a good solution :+1: